collectors/ciscoamp/utils.js

```diff
@@ -0,0 +1,93 @@
+const RestServiceClient = require('@alertlogic/al-collector-js').RestServiceClient;
+var url = require("url");
+
+const Audit_Logs = 'AuditLogs';
+const Events = 'Events';
+
+function getAPILogs(baseUrl, authorization, apiUrl, state, accumulator, maxPagesPerInvocation) {
+    let pageCount = 0;
+    let nextPage;
+    let resetSeconds = null;
+    let totalLogsCount = state.totalLogsCount;
+    let discardFlag = false;
+
+    let restServiceClient = new RestServiceClient(baseUrl);
+
+    return new Promise(function (resolve, reject) {
+        getData();
+        function getData() {
+            if (pageCount < maxPagesPerInvocation) {
+                return restServiceClient.get(apiUrl, {
+                    headers: {
+                        "authorization": `Basic ${authorization}`
+                    },
+                    resolveWithFullResponse: true
+                }).then(({ headers, body }) => {
+                    pageCount++;
+                    if (parseInt(headers['x-ratelimit-remaining']) < 200) {
```

Comment:
Why less than 200?

Reply:
Hey Joe, Actually the hourly limit of request is 3000. After 2800 requests we check that condition and will stop requesting the server.

In AWS testing (Cloudwatch log groups), Every stack log streams AWS Assign an IP address and they are not the same. By IP our request limit is work (Every IP 3000 hourly request is allowed).

I think this condition rarely will work because As we know In hour 3600 seconds and we have 3000 request limit and every collector call at list required  2 to 3  seconds so in an hour max we used 1800 request with one log streams